File: csharp/CsSearch/CsSearchLib/SearchConfig.cs
using System;
using System.IO;
using CsFindLib;

namespace CsSearchLib;

public class SearchConfig : FindConfig
{
    public string SearchOptionsPath { get; private set; }
    public string DefaultSearchSettingsPath { get; private set; }

    public SearchConfig()
    {
        SearchOptionsPath = "CsSearchLib.Resources.searchoptions.json";

        var homePath = FileUtil.GetHomePath();
        var defaultXSearchConfigDir = Path.Join(homePath, ".config", "xsearch");
        var xSearchConfigDir = Environment.GetEnvironmentVariable("XSEARCH_CONFIG_DIR");
        if (string.IsNullOrEmpty(xSearchConfigDir))
        {
            xSearchConfigDir = defaultXSearchConfigDir;
        }

        DefaultSearchSettingsPath = Path.Join(xSearchConfigDir, "settings.json");
    }
}

File: fsharp/FsSearch/FsSearchLib/SearchConfig.fs
namespace FsSearchLib
open System
open System.IO
open FsFindLib

type SearchConfig() =
    inherit FindConfig()

    let getXSearchConfigDir () =
        let defaultXSearchConfigDir = Path.Join(FileUtil.GetHomePath(), ".config", "xsearch")
        let xSearchConfigDir = Environment.GetEnvironmentVariable("XSEARCH_CONFIG_DIR")
        if String.IsNullOrEmpty(xSearchConfigDir) then
            defaultXSearchConfigDir
        else
            xSearchConfigDir

    member val SearchOptionsPath : string = "FsSearchLib.Resources.searchoptions.json" with get, set
    member val DefaultSearchSettingsPath : string = Path.Join(getXSearchConfigDir(), "settings.json") with get, set
